package.json

```
@@ -42,17 +43,19 @@
     "@semantic-release/npm": "^7.0.3",
     "@semantic-release/release-notes-generator": "^9.0.0",
     "@types/eslint": "^6.1.8",
-    "@types/estree": "0.0.42",
+    "@types/jest": "^25.0.0",
     "@types/node": "^13.7.1",
-    "@typescript-eslint/eslint-plugin": "^2.19.2",
-    "@typescript-eslint/parser": "^2.19.2",
+    "@typescript-eslint/eslint-plugin": "^2.34.0",
+    "@typescript-eslint/parser": "^2.34.0",
```

probably should update this to allow the recent v3 release
```suggestion
    "@typescript-eslint/parser": "^2.34.0 || ^3.0.0",
```

---

Should v3 be compatible with rules developed against v2? And would having a dep on `@typescript-eslint/experimental-utils` v2 then pulling the parser v3 in as a peer dep cause problems?

---

> Should v3 be compatible with rules developed against v2

Most of the breaking changes in v3 were deprecations and removals.
From the perspective of this rule - none of the AST changes will affect it.

https://github.com/typescript-eslint/typescript-eslint/releases/tag/v3.0.0

---

> And would having a dep on @typescript-eslint/experimental-utils v2 then pulling the parser v3 in as a peer dep cause problems?

From my experience, people consume our tooling in "lock step", because there's no reason not to upgrade everything at the same time.

We also force a version bump for all packages whenever we release (even if there are no changes) to ensure the version numbers are all kept the same.

That being said - there weren't any changes between v2 and v3 that will cause problems if people do weird things with their versions.

---

Looking at the code, the only usage of the parser in the custom rule was to import a type--which is unnecessary after `ESLintUtils.getRequiredParserServices()`, so I removed the parser from dependencies.

---

Oops forgot that the test uses the parser without explicitly requiring it due to how the API works. 🤦‍♀️  Fixed.